logcat/src/main/java/logcat/Logcat.kt

```diff
@@ -74,15 +73,23 @@ inline fun Any.logcat(
 inline fun logcat(
   tag: String,
   priority: LogPriority = DEBUG,
-  message: () -> String
+  message: MessageSupplier
 ) {
   with(LogcatLogger.logger) {
     if (isLoggable(priority)) {
-      log(priority, tag, message())
+      log(priority, tag, message.get())
     }
   }
 }
 
+/**
+ * Similar to kotlin's Function0<T> interface, or Supplier<T> from java, but compatible on all
+ * android API levels and doesn't reference kotlin stdlib.
```

Comment:
This is an implementation detail that belongs in the commit message, not the code.
```suggestion
 * Function to lazily provide the log message, if it is ever needed.
```

Reply:
Good idea, I moved it to the commit message and used your suggestion instead.